diff --git a/node_modules/react-native-screens/lib/commonjs/components/ScreenStack.js b/node_modules/react-native-screens/lib/commonjs/components/ScreenStack.js
index ebf169c..9c85eee 100644
--- a/node_modules/react-native-screens/lib/commonjs/components/ScreenStack.js
+++ b/node_modules/react-native-screens/lib/commonjs/components/ScreenStack.js
@@ -6,10 +6,12 @@ Object.defineProperty(exports, "__esModule", {
 });
 exports.default = void 0;
 var _react = _interopRequireDefault(require("react"));
+var _reactNative = require("react-native");
 var _contexts = require("../contexts");
 var _warnOnce = _interopRequireDefault(require("warn-once"));
 var _ScreenStackNativeComponent = _interopRequireDefault(require("../fabric/ScreenStackNativeComponent"));
 var _flags = _interopRequireDefault(require("../flags"));
+var _core = require("../core");
 function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
 function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); } // Native components
 const assertGHProvider = (ScreenGestureDetector, goBackGesture) => {
@@ -46,6 +48,9 @@ function ScreenStack(props) {
   });
   assertGHProvider(ScreenGestureDetector, goBackGesture);
   assertCustomScreenTransitionsProps(screensRefs, currentScreenId, goBackGesture);
+  if (!(0, _core.screensEnabled)() || !_core.isNativePlatformSupported) {
+    return /*#__PURE__*/_react.default.createElement(_reactNative.View, rest, children);
+  }
   return /*#__PURE__*/_react.default.createElement(_contexts.RNSScreensRefContext.Provider, {
     value: screensRefs
   }, /*#__PURE__*/_react.default.createElement(ScreenGestureDetector, {
diff --git a/node_modules/react-native-screens/lib/module/components/ScreenStack.js b/node_modules/react-native-screens/lib/module/components/ScreenStack.js
index 95fa10b..acaa629 100644
--- a/node_modules/react-native-screens/lib/module/components/ScreenStack.js
+++ b/node_modules/react-native-screens/lib/module/components/ScreenStack.js
@@ -2,12 +2,14 @@
 
 function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
 import React from 'react';
+import { View } from 'react-native';
 import { GHContext, RNSScreensRefContext } from '../contexts';
 import warnOnce from 'warn-once';
 
 // Native components
 import ScreenStackNativeComponent from '../fabric/ScreenStackNativeComponent';
 import featureFlags from '../flags';
+import { screensEnabled, isNativePlatformSupported } from '../core';
 const assertGHProvider = (ScreenGestureDetector, goBackGesture) => {
   const isGestureDetectorProviderNotDetected = ScreenGestureDetector.name !== 'GHWrapper' && goBackGesture !== undefined;
   warnOnce(isGestureDetectorProviderNotDetected, 'Cannot detect GestureDetectorProvider in a screen that uses `goBackGesture`. Make sure your navigator is wrapped in GestureDetectorProvider.');
@@ -42,6 +44,9 @@ function ScreenStack(props) {
   });
   assertGHProvider(ScreenGestureDetector, goBackGesture);
   assertCustomScreenTransitionsProps(screensRefs, currentScreenId, goBackGesture);
+  if (!screensEnabled() || !isNativePlatformSupported) {
+    return /*#__PURE__*/React.createElement(View, rest, children);
+  }
   return /*#__PURE__*/React.createElement(RNSScreensRefContext.Provider, {
     value: screensRefs
   }, /*#__PURE__*/React.createElement(ScreenGestureDetector, {
